Purchases/Public/IntroEligibility.swift

```diff
@@ -54,17 +56,16 @@ private extension IntroEligibilityStatus {
     }
 
     init(statusCode: Int) throws {
-        switch statusCode {
-        case 0:
-            self = .unknown
-        case 1:
-            self = .ineligible
-        case 2:
-            self = .eligible
-        default:
+        guard let result = Self.mapping[statusCode] else {
             throw IntroEligibilityStatusError.invalidStatusCode(statusCode)
         }
+
+        self = result
     }
+
+    private static let mapping: [Int: IntroEligibilityStatus] = Dictionary(
+        uniqueKeysWithValues: IntroEligibilityStatus.allCases.map { ($0.rawValue, $0) }
+    )
```

Comment:
would using `RawRepresentable` remove the need to do this mapping ourselves?

Reply:
Having an associated `Int` implicitly makes the type `RawRepresentable`. But that doesn't provide this _reverse_ mapping from `rawValue` to the type.